worker-workflow/src/main/resources/add-failures.js

```diff
@@ -40,8 +40,15 @@ function addFailures (document, failures, extractSourceCallback, action) {
             ? extractSourceCallback(f)
             : getCurrentWorkerName(document) + " " + getCurrentWorkerVersion(document);
         var isWarningFlag = (typeof isWarning === 'function') ? isWarning(f): false;
```

Comment:
Make it so that we only call the `isWarning()` function if we need to (i.e. so if the failure id was `ABC-WARNING` then we don't need to).  At the minute we're calling it first.